src/commands/runDraftTool/runDraftDockerfile.ts

```diff
@@ -0,0 +1,330 @@
+import {longRunning} from '../../utils/host';
+import {Context} from './model/context';
+import * as vscode from 'vscode';
+import {downloadDraftBinary, runDraftCommand} from './helper/runDraftHelper';
+import {DraftLanguage, draftLanguages} from './helper/languages';
+import {
+   AzureWizard,
+   AzureWizardExecuteStep,
+   AzureWizardPromptStep,
+   IActionContext
+} from '@microsoft/vscode-azext-utils';
+import * as path from 'path';
+import * as fs from 'fs';
+import {
+   buildCreateCommand,
+   buildCreateConfig
+} from './helper/draftCommandBuilder';
+import {State, StateApi} from '../../utils/state';
+import {PromptPort, ignoreFocusOut} from './helper/commonPrompts';
+
+const title = 'Draft a Dockerfile from source code';
+
+interface PromptContext {
+   sourceCodeFolder: vscode.Uri;
+   language: DraftLanguage;
+   version: string;
+   port: string;
+}
+type WizardContext = IActionContext & Partial<PromptContext>;
+type IPromptStep = AzureWizardPromptStep<WizardContext>;
+type IExecuteStep = AzureWizardExecuteStep<WizardContext>;
+
+export async function runDraftDockerfile(
+   {actionContext, extensionContext}: Context,
+   sourceCodeFolder: vscode.Uri | undefined
+) {
+   const state: StateApi = State.construct(extensionContext);
+
+   // Ensure Draft Binary
+   const downloadResult = await longRunning(`Downloading Draft.`, () =>
+      downloadDraftBinary()
+   );
+   if (!downloadResult) {
+      vscode.window.showErrorMessage('Failed to download Draft');
+      return undefined;
+   }
+
+   const workspaceFolders = vscode.workspace.workspaceFolders;
+   if (!sourceCodeFolder && workspaceFolders && workspaceFolders.length !== 0) {
+      sourceCodeFolder = workspaceFolders[0].uri;
+   }
+   const wizardContext: WizardContext = {
+      ...actionContext,
+      sourceCodeFolder: sourceCodeFolder
+   };
+   const promptSteps: IPromptStep[] = [
+      new PromptSourceCodeFolder(),
```

Comment:
Hi Oliver, during the POC we used multistep wizard, which was custom written by one of us I guess and had control over it, AzureWizardPromptStep is from another npm dependency, is there any benefit we get using this prompt step, I feel like it is all same just a wrapper around the multi-step wizard we previously had during the POC. Maybe I am missing something can you clarify please? Sorry I know you explained this in your design doc, just want to get it clarified. Thanks.

Reply:
Thanks for the review Hari!

The one used during the POC was copy/pasted from [here](https://github.com/microsoft/vscode-extension-samples/tree/main/quickinput-sample) (specifically [this](https://github.com/microsoft/vscode-extension-samples/blob/main/quickinput-sample/src/multiStepInput.ts)). It wasn't something that was custom written by us.

The UI tooling (AzureWizardPrompt and others) comes from the `@microsoft/vscode-azext-utils` library. It's better to consume that library than write the UI tooling ourselves. The consumed library provides a number of benefits over what we had in POC including focus control (whether or not to exit experience if someone clicks away), quick open for selecting files, greatly increased telemetry, better error handling with a report issue button, and more options for each prompt. It's just a better way of handling the wizard and all the major Azure extensions that I've seen use it.